app/controllers/api/v1/accounts/agents_controller.rb

```diff
@@ -9,19 +9,16 @@ def index
     @agents = agents
   end
 
-  def destroy
-    @agent.current_account_user.destroy
-    head :ok
-  end
+  def create; end
 
   def update
-    @agent.update!(agent_params.except(:role))
-    @agent.current_account_user.update!(role: agent_params[:role]) if agent_params[:role]
-    render partial: 'api/v1/models/agent.json.jbuilder', locals: { resource: @agent }
+    @agent.update!(agent_params.slice(:name).compact)
```

Comment:
Are the roles check already available, who can modify the agent's data and he/she is not destroying his own record?

Reply:
@tejaswinichile We have a policy check which only allows admins to access `create`, `edit`, and `destroy` endpoints. `index` is available for all users ( both admins and agents ) in the system.  

- An admin can update the name of the user in the system
- An admin can remove himself from the account 
- An admin can remove other users from the account ( both agents and admins )